coderd/notifications/notifications_test.go

```diff
@@ -151,11 +142,8 @@ func TestSMTPDispatch(t *testing.T) {
 func TestWebhookDispatch(t *testing.T) {
 	t.Parallel()
 
-	// setup
-	if !dbtestutil.WillUsePostgres() {
-		t.Skip("This test requires postgres")
-	}
-	ctx, logger, db := setup(t)
+	// SETUP
+	ctx, logger, db := setupInMemory(t)
```

Comment:
The problem with this approach is that we are not testing against the real database. I'm convinced if it is a truly safe way to go. Which parts of the implementation are missing in dbmem?

Reply:
`dbmem` doesn't actually perform the bulk updates, and along with that comes the retry logic; nor does it handle leases or locking.

There still are some tests which test the real database such as `TestBackpressure`, `TestRetries`, and `TestExpiredLeaseIsRequeued`. Each of those test the full scope of enqueue -> dequeue -> update -> requeue -> update.

In this test specifically, the database is not required to be tested - provided the observed behaviour of having a notification be dispatched to a webhook endpoint is achieved; so it's safe to use a mock database.